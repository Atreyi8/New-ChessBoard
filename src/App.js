
import { useReducer } from "react";
import {ChessBoardContext} from "./store/context"
import { reducer,createInitialState } from "./store/reducer";
import ChessBoard from "./components/ChessBoard/ChessBoard"

function App() {
  const [state, dispatch] = useReducer(reducer, null, createInitialState);
  return (
    //wrapping chessBoard with context api
    <ChessBoardContext.Provider value={{state,dispatch}}>
      <h1>Chessboard</h1>
      <ChessBoard />
    </ChessBoardContext.Provider>

  );
}

export default App;
