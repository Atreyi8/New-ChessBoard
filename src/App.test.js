
import { render } from "@testing-library/react";
import App from "./App";

// Mock the ChessBoard component
jest.mock("./components/ChessBoard/ChessBoard", () => () => <div>ChessBoard Component</div>);

describe("App Component", () => {
  test("renders correctly and matches the snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot(); // Takes a snapshot of the rendered output
  });

  test("renders ChessBoard component and matches snapshot", () => {
    const { container } = render(<App />);
    const chessBoardComponent = container.querySelector("div");
    expect(chessBoardComponent).toMatchSnapshot(); // Matches only the ChessBoard part
  });
});
