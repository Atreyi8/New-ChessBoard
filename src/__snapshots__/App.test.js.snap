// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders ChessBoard component and matches snapshot 1`] = `
<div>
  ChessBoard Component
</div>
`;

exports[`App Component renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <h1>
    Chessboard
  </h1>
  <div>
    ChessBoard Component
  </div>
</DocumentFragment>
`;
